### PR TITLE
Validate User Input when Adding New Item

### LINKED ISSUE #9: 9. As a user, I want to be alerted when I’m entering an empty item or an item that's already on my list.
## Summary
Users shouldn't be able to add an empty item to their list, or add the same item twice. If they try to do this, we need to show them an error message that explains the problem. This way, we’ll prevent some clutter in their lists.

## Acceptance criteria
- [x] Show an error message if the user tries to submit an empty item
- [x] Show an error message if the user tries to submit a new item that is _identical_ to an existing name. For instance, if the list contains `apples` and the user adds `apples`.
- [x] Show an error message if the user tries to submit a new item that matches an existing name with punctuation and casing normalized. For instance, if the list contains `apples` and the user adds `aPples` or `apples,` or `a pples`.
- [x] The user’s original input is saved in the database

##  Notes
- There are a couple of ways to approach this task. If you’re not sure how to begin, _ask your mentors_ for advice!

### PR DESCRIPTION
**Pull request by FuentesNina**
## Description

Users can no longer add an empty item or the same item twice. If they try to do this, they receive an error message that explains the problem. This includes showing an error message if the user tries to submit a new item that matches an existing name with punctuation and casing normalized. For instance, if the list contains apples and the user adds aPples or apples, or a pples.

## Related Issue

Closes #9

## Acceptance Criteria

- [x]  Show an error message if the user tries to submit an empty item
- [x]  Show an error message if the user tries to submit a new item that is identical to an existing name. For instance, if the list contains apples and the user adds apples.
- [x]  Show an error message if the user tries to submit a new item that matches an existing name with punctuation and casing normalized. For instance, if the list contains apples and the user adds aPples or apples, or a pples.
- [x]  The user’s original input is saved in the database

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
| ✓   | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before
Variations of existing items can be added to the shopping list
Screenshot 2023-02-05 at 5 29 02 PM

### After
Entering an empty item
Screenshot 2023-02-05 at 5 31 57 PM

Entering an existing item
Screenshot 2023-02-05 at 5 26 12 PM

Entering an existing item with extra spaces and random punctuation
Screenshot 2023-02-05 at 5 26 42 PM

### CODE REVIEW

File: src/views/AddItem.jsx [hunk 1]
```diff
@@ -23,6 +25,32 @@ export function AddItem({ listToken }) {
 			return;
 		}
 
+		const existsInList = (itemName) => {
+			const ALPHABET = 'abcdefghijklmnopqrstuvwxyz';
```
Comment (jennaly) [inline src/views/AddItem.jsx]: can remove this if using .replace

File: src/views/AddItem.jsx [hunk 2]
```diff
@@ -23,6 +25,32 @@ export function AddItem({ listToken }) {
...
+			let transformedInput = [];
```
Comment (jennaly) [inline src/views/AddItem.jsx]:
```suggestion
			let transformedInput = itemName.replace(/[^a-zA-Z]/g, '');
```

File: src/views/AddItem.jsx [hunk 3]
```diff
@@ -23,6 +25,32 @@ export function AddItem({ listToken }) {
...
+
+			for (let i = 0; i < itemName.length; i++) {
+				let char = itemName[i].toLowerCase();
+
+				if (ALPHABET.includes(char)) {
+					transformedInput.push(char);
+				}
+			}
```
Comment (jennaly) [inline src/views/AddItem.jsx]: if you decide to go with .replace, you can remove this block of logic

File: src/views/AddItem.jsx [hunk 4]
```diff
@@ -23,6 +25,32 @@ export function AddItem({ listToken }) {
...
+
+			const duplicates = data.filter((item) => {
+				return item.name.toLowerCase() === transformedInput.join('');
```
Comment (jennaly) [inline src/views/AddItem.jsx]:
```suggestion
				return item.name.toLowerCase() === transformedInput;
```

**Review by jennaly — COMMENTED** [review]
Great work, your code passes all AC and functionalities are working as they should. I added suggestions for an alternative way to catch punctuations in the itemName string with regular expression. If you decide to go this route, it frees up a bit of space from not having to store the alphabet in a variable and also saves you time from needing to write a loop to manually check if each character in itemName is a letter. Your code to check for duplications also works, great job.

**Comment by neshacascia** [comment]
Just playing around with form validation... Referring back to line 23, I know this code is from a few weeks back, but I would suggest updating it now to check if the length of `itemName` after removing all whitespaces equals 0 since currently, the user can enter multiple spaces and it will still get added to the list.
```suggestion
if (itemName.trim().length === 0) {
	alert('Please specify the name of the item');
	return;
}
```

**Comment by neshacascia** [comment]
Awesome job you two! Code meets the AC! 🥳

**Review by meganesu — APPROVED** [review]
Double-checked that all the ACs are met. Well done @FuentesNina and @kurzbird!

And great suggestions from @jennaly and @neshacascia in the peer reviews 👏

**Pull request merged**